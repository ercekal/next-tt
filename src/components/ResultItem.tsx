import React from 'react';
import styled from 'styled-components'

interface ResultItem {
  artistName: string;
  trackName: string;
  collectionName: string;
}

interface ResultItemProps {
  item: ResultItem;
  key: number;
}

const Div = styled.div`
  display: flex;
  flex-direction: column;
  border: 1px solid blue;
  padding: 5px 0;
  margin: 5px;
`

const ResultItem = ({ item }: ResultItemProps) => {
  const { artistName, trackName, collectionName } = item
  return (
    <Div>
      <p>Artist: {artistName}</p>
      <p>Song Name: {trackName}</p>
      <p>Album: {collectionName}</p>
    </Div>
  );
};

export default ResultItem;