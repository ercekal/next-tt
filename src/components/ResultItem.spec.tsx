import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ResultItem from './ResultItem';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


const props = {
  artistName: 'test artist',
  trackName: 'test track',
  collectionName: 'test album',
}
describe('ResultItem component', () => {
  it('Should be rendered with correct props', () => {
    const wrapper = mount(<ResultItem item={props} key={1} />)
    expect(wrapper.find('.ResultItem').first().text()).toBe('Artist: test artistSong Name: test trackAlbum: test album')
  })
  it('renders correctly', () => {
  const tree = renderer
      .create(<ResultItem item={props} key={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})

