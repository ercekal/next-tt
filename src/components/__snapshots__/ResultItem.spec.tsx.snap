// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem component renders correctly 1`] = `
<div
  className="sc-bdnylx jkDdCw ResultItem"
>
  <p>
    Artist: 
    test artist
  </p>
  <p>
    Song Name: 
    test track
  </p>
  <p>
    Album: 
    test album
  </p>
</div>
`;
